### PR TITLE
 Improved conversion of the "Shape" onnx op

### PR DESCRIPTION
**Pull request by benoitsteiner**

### CODE REVIEW

File: apps/onnx/onnx_converter.cc [hunk 1]
```diff
@@ -554,23 +554,22 @@ Node convert_metadata_node(
...
+      }
+      result.outputs[0].rep() = Halide::cast<int64_t>(num_elements);
     } else {
-        std::vector<Halide::Var> var(1);
-        result.outputs[0].rep(var) = Halide::cast<int64_t>(0);
-        for (int i = 0; i < dims.size(); ++i) {
-            result.outputs[0].rep(i) = Halide::cast<int64_t>(dims[i]);
-        }
-        result.outputs[0].shape.push_back(static_cast<int>(input_shape.size()));
+      Halide::Var var;
```
Comment (steven-johnson) [inline apps/onnx/onnx_converter.cc]: Are we assuming that op == "Shape" here? If so, add `internal_assert`

File: apps/onnx/onnx_converter.cc [hunk 2]
```diff
@@ -554,23 +554,22 @@ Node convert_metadata_node(
...
+      Halide::Expr res = Halide::Expr(static_cast<int64_t>(0));
+      for (int i = 0; i < input_shape.size(); ++i) {
+	res = Halide::select(var == i, Halide::cast<int64_t>(input_shape[i]), res);
```
Comment (steven-johnson) [inline apps/onnx/onnx_converter.cc]: nit: indentation

File: apps/onnx/onnx_converter.cc [hunk 3]
```diff
@@ -554,23 +554,22 @@ Node convert_metadata_node(
...
+      }
+      result.outputs[0].rep(var) = res;
```
Comment (steven-johnson) [inline apps/onnx/onnx_converter.cc]: I don't know the semantics of the 'shape' op, so I can't comment on the correctness of the code here.

File: apps/onnx/onnx_converter.cc (hunk 2 above)
Comment (benoitsteiner) [reply apps/onnx/onnx_converter.cc]: Fixed

File: apps/onnx/onnx_converter.cc (hunk 1 above)
Comment (benoitsteiner) [reply apps/onnx/onnx_converter.cc]: For some reason internal_assert is not available from this file. I've used an exception instead as done in other places.

**Pull request merged**